aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -238,11 +238,18 @@ struct TORCH_CUDA_CPP_API CUDAEvent {
   }
 
   void moveHelper(CUDAEvent&& other) {
-    std::swap(flags_, other.flags_);
-    std::swap(is_created_, other.is_created_);
-    std::swap(was_recorded_, other.was_recorded_);
-    std::swap(device_index_, other.device_index_);
-    std::swap(event_, other.event_);
+    // Transfer ownership of all state from other to this
+    flags_ = other.flags_;
+    is_created_ = other.is_created_;
+    was_recorded_ = other.was_recorded_;
+    external_ = other.external_;
+    device_index_ = other.device_index_;
+    event_ = other.event_;
```

Comment:
Could this possibly prevent cudaEventDestroy() from being called on `this->event_`? I suppose the move-assignment operator could potentially mean that this->event_ already points to a valid event...